doc/cascadia/Keybindings-Arguments.md

```diff
@@ -112,19 +120,25 @@ runtimeclass AppKeyBindings : Microsoft.Terminal.Settings.IKeyBindings
 Becomes:
 
 ```csharp
-interface IKeyBindingArgs {
+interface IActionEventArgs
+{
     Boolean Handled;
 }
+interface IActionArgs { /* Empty */ }
 
-runtimeclass CopyTextEventArgs : IKeyBindingArgs
+runtimeclass CopyTextArgs : IActionArgs
 {
     Boolean CopyWhitespace;
 }
-runtimeclass NewTabEventArgs : IKeyBindingArgs { }
-runtimeclass NewTabWithProfileEventArgs : IKeyBindingArgs
+runtimeclass CopyTextEventArgs : CopyTextArgs, IActionEventArgs { }
+
+runtimeclass NewTabEventArgs : IActionEventArgs { }

```

Comment:
I don't believe this to be necessary.

Why can't we do this?

```
runtimeclass ActionEventArgs {
    bool Handled{ set; get; };
    IActionArgs ActionArgs;
}
```

and make the receiver use `as` or `QueryInterface` to get ActionArgs to be the right type?

That way we _don't_ need to straight-up double the number of classes we have to implement. This lets us just have one class instead of littering fifteen or more extra classes into our activatable class list.
}

Reply:
Because we're saying the TypedEventHandler's second arg type will always be an `IActionEventArgs`, the consumer would need to `as` or `QI` _anyway_ -- except instead of QI-ing to the "NewTabActionArgs" type they'd QI to the "NewTabActionEventArgs" type...

Reply:
Oh. That is an idea, now isn't it. Yea I can try that out.

Reply:
This is an _infinitely_ better idea